tools/diagnostics.sh

```diff
@@ -0,0 +1,194 @@
+#!/bin/sh
+#
+# Usage:
+#    tools/diagnostics.sh [node]
+#
+# Output:
+#    show file contents and filesystem permissions for the most recent chutney
+#    network.
+#
+#    If the argument node is specified, show the file contents for that node,
+#    otherwise, show the contents for the first node (000*).
+#
+#    Always show some network-wide diagnostics and authority diagnostics,
+#    regardless of the chosen node.
+#
+# Examples:
+#    tools/diagnostics.sh
+#    tools/diagnostics.sh 001a
+
+if [ ! -d "$CHUTNEY_PATH" ] || [ ! -x "$CHUTNEY_PATH/chutney" ]; then
+    # looks like a broken path: use the path to this tool instead
+    TOOLS_PATH=$(dirname "$0")
+    CHUTNEY_PATH=$(dirname "$TOOLS_PATH")
+    export CHUTNEY_PATH
+fi
+if [ -d "$PWD/$CHUTNEY_PATH" ] && [ -x "$PWD/$CHUTNEY_PATH/chutney" ]; then
+    # looks like a relative path: make chutney path absolute
+    export CHUTNEY_PATH="$PWD/$CHUTNEY_PATH"
+fi
+
+# Get a working net path
+case "$CHUTNEY_DATA_DIR" in
+  /*)
+    # if an absolute path, then leave as-is
+    # chutney will make this directory automatically if needed
+    ;;
+  *)
+    # if a relative path
+    if [ ! -d "$CHUTNEY_DATA_DIR" ]; then
+        # looks like a broken path: use the chutney path as a base
+        export CHUTNEY_DATA_DIR="$CHUTNEY_PATH/net"
```

Comment:
Does this need to be on two lines for non-bash?

Reply:
I don't think so?
Shellcheck doesn't complain, and we do the same thing in test-network.sh.